bin/varnishd/cache/cache_http.c

```diff
@@ -821,36 +821,158 @@ http_GetHdrField(const struct http *hp, hdr_t hdr,
 
 /*--------------------------------------------------------------------*/
 
+static ssize_t
+http_parse_uint(const char *b, const char **e)
+{
+	ssize_t u;
+	unsigned n;
+
+	u = 0;
+	if (!vct_isdigit(*b))
+		return (-1);
+	for (; vct_isdigit(*b); b++) {
+		if (u > (SSIZE_MAX / 10))
+			return (-1);
```

Comment:
by how this function is used ("too big" cases), this should be `return (-2)`?
But see next comment.

Reply:
Good catch!